### PR TITLE
feat!: split field's `band` into field's `bandPosition` & mark's `width/height: {band: ...}`

### PR DESCRIPTION
**Pull request by kanitw**
Also rename `config.mark.timeUnitBand` -> `timeUnitBandSize`

### CODE REVIEW

**Comment by domoritz** [comment]
Fixes https://github.com/vega/vega-lite/issues/6994

File: src/channeldef.ts [hunk 1]
```diff
@@ -472,14 +475,12 @@ export interface PositionBaseMixins {
 
 export interface BandMixins {
   /**
-   * For rect-based marks (`rect`, `bar`, and `image`), mark size relative to bandwidth of [band scales](https://vega.github.io/vega-lite/docs/scale.html#band), bins or time units. If set to `1`, the mark size is set to the bandwidth, the bin interval, or the time unit interval. If set to `0.5`, the mark size is half of the bandwidth or the time unit interval.
-   *
-   * For other marks, relative position on a band of a stacked, binned, time unit or band scale. If set to `0`, the marks will be positioned at the beginning of the band. If set to `0.5`, the marks will be positioned in the middle of the band.
+   * Relative position on a band of a stacked, binned, time unit or band scale. If set to `0`, the marks will be positioned at the beginning of the band. If set to `0.5`, the marks will be positioned in the middle of the band.
```
Comment (domoritz) [inline src/channeldef.ts:478]:
```suggestion
   * Relative position on a band of a stacked, binned, time unit, or band scale. If set to `0`, the marks will be positioned at the beginning of the band. If set to `0.5`, the marks will be positioned in the middle of the band, etc.
```

**Review by domoritz — APPROVED** [review]
LGTM

File: src/channeldef.ts (hunk 1 above)
Comment (kanitw) [reply src/channeldef.ts:478]: I don't think we need etc at the end.

Comment (domoritz) [reply src/channeldef.ts:478]: Maybe say "and so on." then or start with "For example..."? When I read it I was surprised to only see two examples but the text makes it sound like this is all you can set.

**Comment by kanitw** [comment]
Tests are all passing now.  Since a lot has changed since you approved, it might be worth another review pass, just in case.

**Comment by kanitw** [comment]
oops it's still failing ...

**Review by domoritz — APPROVED** [review]
A bit more test coverage would be good.

**Pull request merged**